src/site/content/en/blog/kapwing/index.md

```diff
@@ -0,0 +1,415 @@
+why, take a look at a screenshot from the editor:
+
+{% Img 
+   src="image/VbsHyyQopiec0718rMq2kTE1hke2/6MqSyhNqFl23wN3GWKbP.png", 
+   alt="Kapwing editor has several templates and capabilities",
+   width="800", height="397"
+%}
+
+This is a YouTube style video, which is common in our app. The user does not
+move very much throughout the clip, so the timelines visual thumbnails are not
+as useful for navigating between sections. On the other hand, the audio
+[waveform](https://en.wikipedia.org/wiki/Waveform) shows peaks and valleys,
+with the valleys typically corresponding to dead time in the recording. If you
+zoom in on the timeline, you'd see more fine grained audio information with
+valleys corresponding to stutters and pauses.
+
+Our user research shows that creators are often guided by these waveforms as
+they splice their content. The web audio API allows us to present this
+information performantly and to update quickly on a zoom or pan of the
+timeline.
+
+The snippet below demonstrates how we do this:
+
+```js
+const getDownsampledBuffer = (idbAsset: IdbAsset) =>
+  decodeMutex.runExclusive(
+    async (): Promise<Float32Array> => {
+      const arrayBuffer = await idbAsset.file.arrayBuffer();
+      const audioContext = new AudioContext();
+      const audioBuffer = await audioContext.decodeAudioData(arrayBuffer);
+
+      const offline = new OfflineAudioContext(
+        audioBuffer.numberOfChannels,
+        audioBuffer.duration * MIN_BROWSER_SUPPORTED_SAMPLE_RATE,
+        MIN_BROWSER_SUPPORTED_SAMPLE_RATE
+      );
+
+      const downsampleSource = offline.createBufferSource();
+      downsampleSource.buffer = audioBuffer;
+      downsampleSource.start(0);
+      downsampleSource.connect(offline.destination);
+
+      const downsampledBuffer22K = await offline.startRendering();
+
+      const downsampledBuffer22KData = downsampledBuffer22K.getChannelData(0);
+
+      const downsampledBuffer = new Float32Array(
+        Math.floor(
+          downsampledBuffer22KData.length / POST_BROWSER_SAMPLE_INTERVAL
+        )
+      );
+
+      for (
+        let i = 0, j = 0;
+        i < downsampledBuffer22KData.length;
+        i += POST_BROWSER_SAMPLE_INTERVAL, j += 1
+      ) {
+        let sum = 0;
+        for (let k = 0; k < POST_BROWSER_SAMPLE_INTERVAL; k += 1) {
+          sum += Math.abs(downsampledBuffer22KData[i + k]);
+        }
+        const avg = sum / POST_BROWSER_SAMPLE_INTERVAL;
+        downsampledBuffer[j] = avg;
+      }
+
+      return downsampledBuffer;
+    } 
+  );
+```
+
+We pass this helper the asset that is stored in indexedDB. Upon completion we 
+will update the asset in indexedDB as well as our own cache.
+
+We gather data about the audioBuffer with the AudioContext constructor, but 
+because we aren't rendering to the device hardware we use the 
+`OfflineAudioContext` to render to an `ArrayBuffer` where we will store 
+amplitude data.
```

Comment:
```suggestion
We gather data about the `audioBuffer` with the `AudioContext` constructor, but 
because we aren't rendering to the device hardware we use the 
`OfflineAudioContext` to render to an `ArrayBuffer` where we will store 
amplitude data.
```